core/embed/trezorhal/stm32u5/secret.c

```diff
@@ -37,10 +37,10 @@ secbool secret_ensure_initialized(void) {
 }
 
 secbool secret_bootloader_locked(void) {
-#ifdef FIRMWARE
-  return (TAMP->BKP8R != 0) * sectrue;
+#if defined BOOTLOADER || defined BOARDLOADER
+  return secret_optiga_present();
 #else
-  return sectrue;
+  return (TAMP->BKP8R != 0) * sectrue;
```

Comment:
To be precise, we should probably test all eight words of the secret for zeros, not just the first one.

Reply:
fixed https://github.com/trezor/trezor-firmware/pull/4157/commits/2d8ed383f9cf4db3d2324b5254a9481a1cd5e69d